### PR TITLE
handle 0 and NULL value of NTH_VALUE function

### LINKED ISSUE #12320: Bug in `nth_value()` window function for `NULL` input (SQLancer)
### Describe the bug

There are two unexcepted behavior in `nth_value()` window function

1. When `NULL` literal is used in the second arg, it returned `Execution error: Expected an integer value`
The expected behavior is to evaluate all rows to `NULL`, because `NULL` should be treated as a missing integer value

2. When second arg is `col*NULL`, it returned another unexpected internal error

More details can be found in the reproducer in datafusion-cli (Compiled from latest main, commit hash 6bbad7eda, with `cargo run`)
```
DataFusion CLI v41.0.0
> create table t1(v1 int, v2 int);
insert into t1 values (1,1), (1,2),(1,3),(2,1),(2,2);

> select v1, nth_value(v2, null) over (partition by v1 order by v2) from t1;

Execution error: Expected an integer value

> select v1, nth_value(v2, v2*null) over (partition by v1 order by v2) from t1;
Internal error: Cannot convert Int64(NULL) to i64.
This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
```

Expected behavior in DuckDB (output in postgres is the same)
```
D select v1, nth_value(v2, null) over (partition by v1 order by v2) from t1;
┌───────┬────────────────────────────────────────────────────────┐
│  v1   │ nth_value(v2, NULL) OVER (PARTITION BY v1 ORDER BY v2) │
│ int32 │                         int32                          │
├───────┼────────────────────────────────────────────────────────┤
│     2 │                                                        │
│     2 │                                                        │
│     1 │                                                        │
│     1 │                                                        │
│     1 │                                                        │
└───────┴────────────────────────────────────────────────────────┘

D select v1, nth_value(v2, v2*null) over (partition by v1 order by v2) from t1;
┌───────┬───────────────────────────────────────────────────────────────┐
│  v1   │ nth_value(v2, (v2 * NULL)) OVER (PARTITION BY v1 ORDER BY v2) │
│ int32 │                             int32                             │
├───────┼───────────────────────────────────────────────────────────────┤
│     2 │                                                               │
│     2 │                                                               │
│     1 │                                                               │
│     1 │                                                               │
│     1 │                                                               │
└───────┴───────────────────────────────────────────────────────────────┘
```

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

Found by SQLancer https://github.com/apache/datafusion/issues/11030

### PR DESCRIPTION
**Pull request by thinh2**
## Which issue does this PR close?

Closes #12320.

## Rationale for this change

The root cause in #12320 is because `get_signed_integer_value` will return an error on `NULL` or `Int64(NULL)` value. To process the `nth_value` function, it requires the `n` value should be an integer. Therefore, I decided to convert `NULL` value to 0 in `get_signed_integer_value` and implement handler for the `0` case. This will evaluate to NULL value for the `nth_value(v1, NULL)` expression.

Furthermore, it also evaluate the `nth_value(v1, 0)` to NULL instead of generating panic, which match the expected behaviour in duckdb.

## What changes are included in this PR?
[x] Return 0 for NULL value in `get_signed_integer`
[x] Handle nth_value execution when n is 0
[x] Test file updated to test `nth_value(v2, 0)`, `nth_value(v2, NULL)` and `nth_value(v2, v2 * NULL)`.

## Are these changes tested?
Yes

### CODE REVIEW

**Review by jayzhan211 — APPROVED** [review]
👍

File: datafusion/physical-plan/src/windows/mod.rs [hunk 1]
```diff
@@ -185,20 +185,30 @@ fn get_scalar_value_from_args(
...
     value.cast_to(&DataType::Int64)?.try_into()
 }
 
 fn get_unsigned_integer(value: ScalarValue) -> Result<u64> {
+    if value.is_null() {
+        return Ok(0);
+    }
+
     if !value.data_type().is_integer() {
         return Err(DataFusionError::Execution(
```
Comment (comphead) [inline datafusion/physical-plan/src/windows/mod.rs:207]:
```suggestion
        return exec_err!(
```

Comment (thinh2) [reply datafusion/physical-plan/src/windows/mod.rs]: thank @comphead, I already updated it.

**Review by comphead — APPROVED** [review]
lgtm thanks @thinh2

**Pull request merged**